plugin/trino-ranger/src/main/java/io/trino/plugin/ranger/RangerSystemAccessControl.java

```diff
@@ -0,0 +1,713 @@
+        }
+        finally {
+            deactivatePluginClassLoader();
+        }
+        return viewExpression;
+    }
+
+    @Override
+    public List<ViewExpression> getColumnMasks(SystemSecurityContext context, CatalogSchemaTableName tableName, String columnName, Type type)
+    {
+        List<ViewExpression> viewExpression;
+        try {
+            activatePluginClassLoader();
+            viewExpression = systemAccessControlImpl.getColumnMasks(context, tableName, columnName, type);
+        }
+        finally {
+            deactivatePluginClassLoader();
+        }
+        return viewExpression;
+    }
+
+    @Override
+    public void checkCanSetUser(Optional<Principal> principal, String userName)
+    {
+        try {
+            activatePluginClassLoader();
+            systemAccessControlImpl.checkCanSetUser(principal, userName);
+        }
+        finally {
+            deactivatePluginClassLoader();
+        }
+    }
+
+    @Override
+    public void checkCanGrantExecuteFunctionPrivilege(SystemSecurityContext context, String functionName, TrinoPrincipal grantee, boolean grantOption)
+    {
+        try {
+            activatePluginClassLoader();
+            systemAccessControlImpl.checkCanGrantExecuteFunctionPrivilege(context, functionName, grantee, grantOption);
+        }
+        finally {
+            deactivatePluginClassLoader();
+        }
+    }
+
+    @Override
+    public void checkCanSetSchemaAuthorization(SystemSecurityContext context, CatalogSchemaName schema, TrinoPrincipal principal)
+    {
+        try {
+            activatePluginClassLoader();
+            systemAccessControlImpl.checkCanSetSchemaAuthorization(context, schema, principal);
+        }
+        finally {
+            deactivatePluginClassLoader();
+        }
+    }
+
+    @Override
+    public void checkCanShowCreateSchema(SystemSecurityContext context, CatalogSchemaName schemaName)
+    {
+        try {
+            activatePluginClassLoader();
+            systemAccessControlImpl.checkCanShowCreateSchema(context, schemaName);
+        }
+        finally {
+            deactivatePluginClassLoader();
+        }
+    }
+
+    @Override
+    public void checkCanExecuteProcedure(SystemSecurityContext systemSecurityContext, CatalogSchemaRoutineName procedure)
+    {
+        try {
+            activatePluginClassLoader();
+            systemAccessControlImpl.checkCanExecuteProcedure(systemSecurityContext, procedure);
+        }
+        finally {
+            deactivatePluginClassLoader();
+        }
+    }
+
+    @Override
+    public void checkCanExecuteFunction(SystemSecurityContext systemSecurityContext, String functionName)
+    {
+        try {
+            activatePluginClassLoader();
+            systemAccessControlImpl.checkCanExecuteFunction(systemSecurityContext, functionName);
+        }
+        finally {
+            deactivatePluginClassLoader();
+        }
+    }
+
+    private void activatePluginClassLoader()
```

Comment:
What is this for? Should you use Plugin thread safe entities, like `io.trino.plugin.base.classloader.ClassLoaderSafeConnectorAccessControl`?

Reply:
@dprophet

Reply:
This is for Ranger plugins, not Trino plugins. All ranger plugins use this same setup. I kept it there because I will allow switching access control systems based on catalogs. In some cases I need a simple REGEX matching based on LDAP/AD not configured in Ranger.

Reply:
I'm not an expert on Ranger, but I've worked on connecting another service to Ranger in this style (having the implementation be part of the service code rather than part of Ranger).  So this is my understanding of Ranger class loading and other intricacies:

Ranger has its own class loader which is related to the way it handles injecting code into Trino (or other services). To clarify, am I talking about the cases when a plugin is built in Ranger and then added to the classpath of another service - the way it's done in this PR is different. 

In short, Ranger provides a shim class with no logic in it that implements `TrinoAuthorizer` and calls out to another implementation class. The implementation class should be only loaded by `RangerClassLoader` so it can bring in its own dependencies. There is a disconnect between service-specific work (in this case, reading configuration for example) and authorization work.

All that `activatePluginClassLoader` does is:
```java
   public void activate() {
        preActivateClassLoader.set(Thread.currentThread().getContextClassLoader());

        Thread.currentThread().setContextClassLoader(this);
    }
```

In this case, it ~might~ work to remove `RangerClassLoader` completely and unite the current shim/implementation architecture since Trino has its own system for handling plugin dependencies.